trucks.js

```diff
@@ -246,9 +252,47 @@ var foodTrucks = [
   }
 ];
 
-/* 
+var trucks = (function(){
+  var truckName = '';
+  var trucks = [];
+
+  function getTruckByDay( truck, day ){
+    if (truck.schedule.indexOf(day) !== -1){
+      return true;
+    } else {
+      return false;
+    }}
+
+  function filterTrucksByDay( day ){
+    if( day !== null ){
+      trucks = foodTrucks.filter(function( truck ){
+        return getTruckByDay( truck, day );
+      });
+
+      for( var x=0; x<trucks.length; x++ ){
```

Comment:
Better to return the list of truck objects rather than a list of names, as it assumes too much about the intent of the caller.